src/fandom-wirewax.plugin.js

```diff
@@ -0,0 +1,257 @@
+    .catch(function(error) {
+      console.warn(error);
+    });
+  }.bind(this));
+}
+
+FandomWirewaxPlugin.prototype.setupEmbedder = function () {
+  if (!this.embedder) {
+    // Create a container
+    this.container = document.createElement("div");
+    this.container.classList.add("vjs-wirewax-container");
+    this.container.setAttribute(
+      "style",
+      "position: absolute; height: 100%; width: 100%; top: 0; pointer-events: none"
+    );
+    this.player.getContainer().appendChild(this.container);
+
+    // Initialize embedder
+    this.embedder = window.createWirewaxEmbedder();
+  }
+
+  // Create video-less WIREWAX player
+  this.embedder.createEl(this.container, {
+    isPlugin: true,
+    videoId: this.vidId,
+    rootId: this.rootId,
+  });
+
+  return this.embedder.ready();
+}
+
+FandomWirewaxPlugin.prototype.registerEvents = function () {
+  // Custom time sync handler
+  var HTML5VideoEl = this.player.getConfig().mediaElement;
+  this.setWIREWAXCurrentTime = function () {
+    this.embedder.setCurrentTime(HTML5VideoEl.currentTime);
+    this.animationId = window.requestAnimationFrame(
+      this.setWIREWAXCurrentTime
+    );
+  }.bind(this);
+
+  // Handle the delay caused by injecting script
+  var isPlaying = this.player.getState() === "playing";
+
+  if (isPlaying || this.autoPlay) {
+    this.startTimeUpdate();
+    this.embedder.play();
+  }
+
+  if (this.isPlayerRegistered) return;
+
+  window.dataLayer = window.dataLayer || [];
+
+  // Bind WIREWAX to JW player events
+  this.player.on("play", this.JWPlayHandler);
+  this.player.on("pause", this.JWPauseHandler);
+  this.player.on("seek", this.JWSeekHandler);
+
+  // Bind JW to WIREWAX events
+  this.embedder.on("play", this.WirewaxPlayHandler);
+  this.embedder.on("pause", this.WirewaxPauseHandler);
+  this.embedder.on("seeked", this.WirewaxSeekedHandler);
+  this.embedder.on("overlayshow", this.WirewaxOverlayShowHandler);
+  this.embedder.on("overlayhide", this.WirewaxOverlayHideHandler);
+  this.embedder.on("hotspotclick", this.WirewaxHotspotClickHandler);
+  this.embedder.on('overlayevent', this.WirewaxOverlayEventHandler);
+
+  this.isPlayerRegistered = true;
+}
+
+FandomWirewaxPlugin.prototype.startTimeUpdate = function () {
+  window.cancelAnimationFrame(this.animationId);
+  this.animationId = window.requestAnimationFrame(this.setWIREWAXCurrentTime);
+}
+
+FandomWirewaxPlugin.prototype.stopTimeUpdate = function () {
+  window.cancelAnimationFrame(this.animationId);
+}
+
+FandomWirewaxPlugin.prototype.JWPlayHandler = function (event) {
+  this.startTimeUpdate();
+
+  try {
+    this.embedder.play();
+    window.dataLayer.push({ event: 'wirewax', action: 'play', eventData: event });
+  } catch (error) {
+    console.warn(error);
+  }
+};
+
+FandomWirewaxPlugin.prototype.JWPauseHandler = function (event) {
+  this.stopTimeUpdate();
+
+  try {
+    this.embedder.pause();
+    window.dataLayer.push({ event: 'wirewax', action: 'pause', eventData: event });
```

Comment:
This is a great example of an area that would benefit from having the function we created in pathfinder. (later though)